code/player/controller/DefaultWalkController.cs

```diff
@@ -43,9 +52,75 @@ public override void Simulate()
                 SprintSpeed = (MaxSprintSpeed - DefaultSpeed) / player.MaxStamina * player.Stamina + DefaultSpeed;
             }
 
+            IsDiving = Pawn.WaterLevel.Fraction == 1f;
+
+            if (!IsDiving)
+            {
+                TimeSinceDivingStarted = 0f;
+            }
+            else if (Host.IsServer && TimeSinceDivingStarted > DrownDamageTime)
+            {
+                using (Prediction.Off())
+                {
+                    DamageInfo damageInfo = new DamageInfo
+                    {
+                        Attacker = Pawn,
+                        Flags = DamageFlags.Drown,
+                        HitboxIndex = (int) HitboxIndex.Head,
+                        Position = Position,
+                        Damage = MathF.Max(DrownDamagePerSecond * Time.Delta, 0f)
+                    };
+
+                    Pawn.TakeDamage(damageInfo);
+                }
```

Comment:
Should prediction be turned off for damage related things? Could this cause issues? This might be a case where we want to leave it on.

Reply:
As damage shouldn't be predictable (or am I wrong?) and shouldn't be send multiple times, I wouldn't turn it on here